crates/primitives/snos-output/src/tests.rs

```diff
@@ -16,7 +16,7 @@ fn segment_decode<T: Decode + SnosCodec, I: Read>(input: &mut I) -> Vec<T> {
     let mut items: Vec<T> = Vec::new();
     while segment_len > 0 {
         let item = T::decode(input);
-        segment_len -= item.felt_len();
+        segment_len -= item.size_hint();
```

Comment:
I think you should be using the actual item size instead of `size_hint`.
For parity_scale_codec, `size_hint` is just a hint here to optimize memory allocation, not something you should rely on 100%. It can be an erroneous value.

Reply:
Or you can change this for your trait. But you should rename the method to not be a `hint` anymore and explain in the doc how the user should impl it

Reply:
That's why I named it differently :) ok